parser/frontmatter_test.go

```diff
@@ -14,9 +14,210 @@
 package parser
 
 import (
+	"bytes"
+	"reflect"
 	"testing"
 )
 
+func TestInterfaceToConfig(t *testing.T) {
+	cases := []struct {
+		input interface{}
+		mark  rune
+		want  []byte
+		isErr bool
+	}{
+		// TOML
+		{map[string]interface{}{}, '+', nil, false},
+		{
+			map[string]interface{}{"title": "test 1"},
+			'+',
+			[]byte("title = \"test 1\"\n"),
+			false,
+		},
+
+		// YAML
+
+		// FIXME(moorereason): ?? returns "{}\n"
```

Comment:
The following commented-out test for an empty map returns `{}\n` for YAML conversion when I expect it to return `nil`.  Didn't take time to look into it before, but YAML apparently can define an empty map as `{}`.  I'd prefer that the results of each format all return a `nil` slice in that situation.  What do you think?  Do we check for `[]byte("{}\n")` and return `nil`?

```go
// {map[string]interface{}{}, '-', nil, false},
```